packages/integrations/src/reportingobserver.ts

```diff
@@ -40,66 +42,16 @@ interface InterventionReportBody {
   columnNumber?: number;
 }
 
-const SETUP_CLIENTS: Client[] = [];
-
-/** Reporting API integration - https://w3c.github.io/reporting/ */
-export class ReportingObserver implements Integration {
-  /**
-   * @inheritDoc
-   */
-  public static id: string = 'ReportingObserver';
-
-  /**
-   * @inheritDoc
-   */
-  public readonly name: string;
-
-  private readonly _types: ReportTypes[];
-
-  /**
-   * @inheritDoc
-   */
-  public constructor(
-    options: {
-      types?: ReportTypes[];
-    } = {},
-  ) {
-    this.name = ReportingObserver.id;
-
-    this._types = options.types || ['crash', 'deprecation', 'intervention'];
-  }
-
-  /**
-   * @inheritDoc
-   */
-  public setupOnce(_: (callback: EventProcessor) => void, _getCurrentHub: () => Hub): void {
-    if (!supportsReportingObserver()) {
-      return;
-    }
-
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
-    const observer = new (WINDOW as any).ReportingObserver(this.handler.bind(this), {
-      buffered: true,
-      types: this._types,
-    });
-
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-    observer.observe();
-  }
+interface ReportingObserverOptions {
+  types?: ReportTypes[];
+}
 
-  /** @inheritdoc */
-  public setup(client: Client): void {
-    if (!supportsReportingObserver()) {
-      return;
-    }
+const SETUP_CLIENTS: Client[] = [];
```

Comment:
IMO this should be a weak map so it can still get GC'd

Reply:
good point!